.github/workflows/test-action.yml

```diff
@@ -94,3 +94,136 @@ jobs:
+            exit 1
+          fi
+
+          # Verify repo matches expected format
+          if [ "${{ steps.test-ci-vars.outputs.repo }}" != "aaronsteers/resolve-vars-action" ]; then
+            echo "ERROR: repo should be aaronsteers/resolve-vars-action"
+            echo "Actual: ${{ steps.test-ci-vars.outputs.repo }}"
+            exit 1
+          fi
+
+          # Verify repo-owner matches expected value
+          if [ "${{ steps.test-ci-vars.outputs.repo-owner }}" != "aaronsteers" ]; then
+            echo "ERROR: repo-owner should be aaronsteers"
+            echo "Actual: ${{ steps.test-ci-vars.outputs.repo-owner }}"
+            exit 1
+          fi
+
+          # Verify repo-name matches expected value
+          if [ "${{ steps.test-ci-vars.outputs.repo-name }}" != "resolve-vars-action" ]; then
+            echo "ERROR: repo-name should be resolve-vars-action"
+            echo "Actual: ${{ steps.test-ci-vars.outputs.repo-name }}"
+            exit 1
+          fi
+
+          # Verify repo-name-full matches expected value
+          if [ "${{ steps.test-ci-vars.outputs.repo-name-full }}" != "aaronsteers/resolve-vars-action" ]; then
+            echo "ERROR: repo-name-full should be aaronsteers/resolve-vars-action"
+            echo "Actual: ${{ steps.test-ci-vars.outputs.repo-name-full }}"
+            exit 1
+          fi
+
+          # Verify restricted-security-mode is false for non-fork PRs
+          if [ "${{ steps.test-ci-vars.outputs.restricted-security-mode }}" != "false" ]; then
+            echo "ERROR: restricted-security-mode should be false for non-fork PRs"
+            echo "Actual: ${{ steps.test-ci-vars.outputs.restricted-security-mode }}"
+            exit 1
+          fi
+
+          # Check that custom variables are still included
+          if [ "${{ fromJSON(steps.test-ci-vars.outputs.all).custom_var }}" != "test_value" ]; then
+            echo "ERROR: custom_var should still be included with standard CI vars"
+            exit 1
+          fi
+
+          echo "✅ Standard CI variables test passed!"
+
+      - name: Test standard CI vars disabled (backward compatibility)
+        id: test-no-ci-vars
+        uses: ./
+        with:
+          standard_ci_vars: false
+          static_inputs: |
+            test_var=backward_compat
+
+      - name: Verify backward compatibility
+        run: |
+          # Ensure CI variables are NOT populated when disabled
+          if [ -n "${{ steps.test-no-ci-vars.outputs.repo }}" ]; then
+            echo "ERROR: repo should not be populated when standard_ci_vars is false"
+            exit 1
+          fi
+
+          # Ensure custom variables still work
+          if [ "${{ fromJSON(steps.test-no-ci-vars.outputs.all).test_var }}" != "backward_compat" ]; then
+            echo "ERROR: custom variables should still work when standard_ci_vars is false"
+            exit 1
+          fi
+
+          echo "✅ Backward compatibility test passed!"
```

Comment:
This step can then be renamed to just 'Test with standard CI vars disabled'. (I don't like "backwards compatibility" as an evergreen name.)